OrbitLinuxTracing/UprobesReturnAddressManagerTest.cpp

```diff
@@ -390,4 +392,228 @@ TEST(UprobesReturnAddressManager, TailCallOptimization) {
+    "7f075b800000-7f075b803000 r--p 00000000 fe:00 2112042                    "
+    "/usr/lib/x86_64-linux-gnu/libgcc_s.so.1\n"
+    "7f075b803000-7f075b814000 r-xp 00003000 fe:00 2112042                    "
+    "/usr/lib/x86_64-linux-gnu/libgcc_s.so.1\n"
+    "7f075b814000-7f075b818000 r--p 00014000 fe:00 2112042                    "
+    "/usr/lib/x86_64-linux-gnu/libgcc_s.so.1\n"
+    "7f075b818000-7f075b819000 r--p 00017000 fe:00 2112042                    "
+    "/usr/lib/x86_64-linux-gnu/libgcc_s.so.1\n"
+    "7f075b819000-7f075b81a000 rw-p 00018000 fe:00 2112042                    "
+    "/usr/lib/x86_64-linux-gnu/libgcc_s.so.1\n"
+    "7f075b81a000-7f075b8b0000 r--p 00000000 fe:00 2112089                    "
+    "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28\n"
+    "7f075b8b0000-7f075b98b000 r-xp 00096000 fe:00 2112089                    "
+    "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28\n"
+    "7f075b98b000-7f075b9d4000 r--p 00171000 fe:00 2112089                    "
+    "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28\n"
+    "7f075b9d4000-7f075b9d5000 ---p 001ba000 fe:00 2112089                    "
+    "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28\n"
+    "7f075b9d5000-7f075b9e0000 r--p 001ba000 fe:00 2112089                    "
+    "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28\n"
+    "7f075b9e0000-7f075b9e3000 rw-p 001c5000 fe:00 2112089                    "
+    "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28\n"
+    "7f075b9e3000-7f075b9e6000 rw-p 00000000 00:00 0 \n"
+    "7f075ba00000-7f075ba02000 rw-p 00000000 00:00 0 \n"
+    "7f075ba02000-7f075ba03000 r--p 00000000 fe:00 2131077                    "
+    "/usr/lib/x86_64-linux-gnu/ld-2.29.so\n"
+    "7f075ba03000-7f075ba22000 r-xp 00001000 fe:00 2131077                    "
+    "/usr/lib/x86_64-linux-gnu/ld-2.29.so\n"
+    "7f075ba22000-7f075ba2a000 r--p 00020000 fe:00 2131077                    "
+    "/usr/lib/x86_64-linux-gnu/ld-2.29.so\n"
+    "7f075ba2a000-7f075ba2b000 r--p 00027000 fe:00 2131077                    "
+    "/usr/lib/x86_64-linux-gnu/ld-2.29.so\n"
+    "7f075ba2b000-7f075ba2c000 rw-p 00028000 fe:00 2131077                    "
+    "/usr/lib/x86_64-linux-gnu/ld-2.29.so\n"
+    "7f075ba2c000-7f075ba2d000 rw-p 00000000 00:00 0 \n"
+    "7ffcae624000-7ffcae646000 rw-p 00000000 00:00 0                          "
+    "[stack]\n"
+    "7ffcae7f0000-7ffcae7f3000 r--p 00000000 00:00 0                          "
+    "[vvar]\n"
+    "7ffcae7f3000-7ffcae7f4000 r-xp 00000000 00:00 0                          "
+    "[vdso]\n"
+    "7fffffffe000-7ffffffff000 --xp 00000000 00:00 0                          "
+    "[uprobes]";
+
+auto maps = LibunwindstackUnwinder::ParseMaps(maps_string);
+
+TEST(UprobesReturnAddressManager, CallchainNoUprobes) {
+  UprobesReturnAddressManager return_address_manager;
+
+  std::vector<uint64_t> expected_callchain{
+      18446744073709551104lu, 94355907990078lu, 94355907990120lu,
```

Comment:
Soooo, since these values are "real" instruction pointer values, why didn't you use the hexadecimal notation? That way it would be possible to make a parallel with the maps, while like this they look like random numbers.

Reply:
Converted all values to hex.